src/rules/html/ensureRequiredAttributes.js

```diff
@@ -0,0 +1,46 @@
+import { HTML_TAGS_WITH_REQUIRED_ATTRIBUTES } from 'const';
+import * as messages from 'messages';
+
+
+export var ensureRequiredAttributes = ($, filename=null) => {
+  return new Promise((resolve, reject) => {
+    var promises = [];
+    var validatorMessages = [];
+
+    for (let tag in HTML_TAGS_WITH_REQUIRED_ATTRIBUTES) {
+      promises.push(_ensureAttributesInTag($, tag,
+        HTML_TAGS_WITH_REQUIRED_ATTRIBUTES[tag], validatorMessages, filename));
+    }
+
+    Promise.all(promises)
+      .then(() => {
+        resolve(validatorMessages);
+      })
+      .catch(reject);
+  });
+};
+
+var _ensureAttributesInTag = (
+($, tag, attributes, validatorMessages, filename=null) => {
+  return new Promise((resolve) => {
+    $('prefwindow').each((i, element) => {
+      for (let attributeName in attributes) {
+        var errorCode = `${tag}_REQUIRES_${attributeName}`.toUpperCase();
+
+        if ($(element).attr(attributeName) === undefined) {
+          validatorMessages.push({
+            code: errorCode,
+            message: messages[errorCode].message,
+            description: messages[errorCode].description,
+            sourceCode: `<${tag}>`,
+            file: filename,
+            // TODO: This is where we store severity. Make it nicer?
+            type: attributes[attributeName],
```

Comment:
It is, but the way it’s stored is bad (see the TODO above). I’ll clean that up, good call.